src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -3068,6 +3068,118 @@ public override MemberInfo[] GetMember(string name, MemberTypes type, BindingFla
 
             return compressMembers;
         }
+
+        public override MemberInfo? GetMemberWithSameMetadataDefinitionAs(MemberInfo member)
+        {
+            if (member is null) throw new ArgumentNullException(nameof(member));
+
+            return member.MemberType switch
+            {
+                MemberTypes.Method => GetMethodWithSameMetadataDefinitionAs(member),
+                MemberTypes.Constructor => GetConstructorWithSameMetadataDefinitionAs(member),
+                MemberTypes.Property => GetPropertyWithSameMetadataDefinitionAs(member),
+                MemberTypes.Field => GetFieldWithSameMetadataDefinitionAs(member),
+                MemberTypes.Event => GetEventWithSameMetadataDefinitionAs(member),
+                MemberTypes.NestedType => GetNestedTypeWithSameMetadataDefinitionAs(member),
```

Comment:
Is this going to do anything interesting for nested types? The nested types are not instantiated using the enclosed type instantiation, so I think this will always return the same type as that was passed in.

Reply:
It probably isn't going to do anything interesting, but I thought it would be incomplete without it, since `GetMembers` returns the nested types.

> The nested types are not instantiated using the enclosed type instantiation,

I learned that the hard way in my tests 😄 

https://github.com/dotnet/runtime/blob/7a15f9750f654d362b452e0ba5251e9756c16860/src/libraries/System.Reflection/tests/TypeInfoTests.cs#L1581-L1584